File: INLBreadcrumbs/INLNavbarTitleView.m
//
//  INLNavbarTitleView.m
//  INLBreadcrumbs
//
//  Created by Tomas Hakel on 02/12/2015.
//  Copyright © 2015 Inloop, s.r.o. See LICENSE.md
//

#import "INLNavbarTitleView.h"

@implementation INLNavbarTitleView

-(void)resize {
	[self.titleLabel sizeToFit];
	self.frame = CGRectMake(self.frame.origin.x, self.frame.origin.y, self.titleLabel.frame.size.width, self.frame.size.height);
}

- (IBAction)titleTapped:(id)sender {
	if (self.onTitleTapped) {
		self.onTitleTapped();
	}
}

-(void)setTitle:(NSString *)title {
	self.titleLabel.text = title;
}

-(NSString *)title {
	return self.titleLabel.text;
}

@end

File: INLBreadcrumbs/INLNavbarTitleView.swift
//
//  INLNavbarTitleView.swift
//  INLBreadcrumbs
//
//  Created by Petr Tomicek on 13.12.16.
//  Copyright © 2016 Inloop, s.r.o. All rights reserved.
//

import UIKit

class INLNavbarTitleView: UIView {
    
    @IBOutlet weak var titleLabel: UILabel!
    
    var titleTappedDelegate: (() -> Void)?
    
    var title: String? {
        
        didSet {
            
            titleLabel.text = title
        }
    }
    
    func resize() {
        
        titleLabel.sizeToFit()
        self.frame = CGRect(x: self.frame.origin.x,
                            y: self.frame.origin.y,
                            width: self.titleLabel.frame.size.width,
                            height: self.frame.size.height)
    }
    
    @IBAction func titleTapped(_ sender: Any)
    {
        titleTappedDelegate?()
    }
    
}

